packages/network-controller/src/NetworkController.ts

```diff
@@ -896,6 +1156,246 @@ export class NetworkController extends BaseControllerV2<
       };
     });
   }
-}
 
-export default NetworkController;
+  /**
+   * Before accessing or switching the network, the registry of network clients
+   * needs to be populated. Otherwise, `#applyNetworkSelection` and
+   * `getNetworkClients` will throw an error. This method checks to see if the
+   * population step has happened yet, and if not, makes it happen.
+   *
+   * @returns The populated network client registry.
+   */
+  #ensureAutoManagedNetworkClientRegistryPopulated(): AutoManagedNetworkClientRegistry {
+    const autoManagedNetworkClientRegistry =
+      this.#autoManagedNetworkClientRegistry ??
+      this.#createAutoManagedNetworkClientRegistry();
+    this.#autoManagedNetworkClientRegistry = autoManagedNetworkClientRegistry;
+    return autoManagedNetworkClientRegistry;
+  }
+
+  /**
+   * Constructs the registry of network clients based on the set of built-in
+   * networks as well as the custom networks in state.
+   *
+   * @returns The network clients keyed by ID.
+   */
+  #createAutoManagedNetworkClientRegistry(): AutoManagedNetworkClientRegistry {
+    return [
+      ...this.#buildIdentifiedInfuraNetworkClientConfigurations(),
+      ...this.#buildIdentifiedCustomNetworkClientConfigurations(),
+      this.#buildIdentifiedNetworkClientConfigurationFromProviderConfig(),
+    ].reduce(
+      (
+        obj,
```

Comment:
Nit: Better to have a more descriptive name if possible, e.g. `registry`

Reply:
Sure, makes sense. Changed here: [6977c44](https://github.com/MetaMask/core/pull/1439/commits/6977c44277fa8e914c860b46751a8678277c85a9)